### PR TITLE
chore: ARM-282 - Added tests to dialog component in module-new

### PR DESCRIPTION
**Pull request by jlews**
## What's new?

- Added tests to dialog component in module-new

## Ticket number(s) in JIRA (if internal)

ARM-282

[board](https://rocketmakers.atlassian.net/jira/software/projects/HM/boards/172)

## Checklist

- [x] are your changes in Storybook?
- [x] are any breaking changes documented in `docs/migrating_from_oldstrong.md`?
- [x] does _everything_ have jsdoc?
- [x] is everything exported from index.ts?
- [x] are all new hooks added to `src/hooks/hooks.stories.mdx` or given their own docs in Storybook?
- [x] are all new SCSS mixins added to `src/theme/mixins.stories.mdx`?
- [x] are all new SCSS variables added to `src/theme/variables.stories.mdx`?

### CODE REVIEW

File: module-new/src/components/dialog/dialog.stories.tsx [hunk 1]
```diff
@@ -22,51 +24,124 @@ const Template: StoryObj<typeof Dialog> = {
...
+          Open dialog
+        </Button>
+        <Dialog testId="dialog-component" {...args} open={dialogOpen} onOpenChange={setDialogOpen} />
       </>
     );
   },
 };
 
+const testTitle = 'Test Dialog';
+const testDesc = 'Hello, I am a test dialog';
```
Comment (judithrocketmakers) [inline module-new/src/components/dialog/dialog.stories.tsx:37]: If you're using these consistently, you can set them as args in your Template and they'll spread onto the other components. For the no title test, I think you can just put a replacement args block in.

File: module-new/src/components/dialog/dialog.component.tsx [hunk 2]
```diff
@@ -73,6 +75,7 @@ export const Dialog = React.forwardRef(
       className,
       data,
       overlayClassName,
+      testId,
```
Comment (judithrocketmakers) [inline module-new/src/components/dialog/dialog.component.tsx:78]: So, testId... It's one of those testing patterns that keeps falling in and out of favour, and I think we're currently in an out cycle. The ideal at the moment is 'write your tests around what the user can see and don't clutter your data-* tags', so in this case unless there's a good reason not to the recommendation would be to use `getByRole('dialog')` as your matcher.

That said, it's almost certainly going to swing back the other way in n years, and I personally think having it as an option on the component for when one can't target it in any other way isn't a bad thing, so... -shrugs expansively-

I think my suggestion overall is to keep it but to try not to use it in your own tests as a good example?

Comment (judithrocketmakers) [reply module-new/src/components/dialog/dialog.component.tsx:78]: See also 'getByText' for the buttons etc.

Comment (jlews) [reply module-new/src/components/dialog/dialog.component.tsx:78]: I suppose that sort of makes sense... Presumably if there's multiple buttons/inputs on screen and I want to click/type into a specific one, a testId is the only way? There's not always something else to discriminate them

Comment (jlews) [reply module-new/src/components/dialog/dialog.component.tsx:78]: All testIds now removed in favour of more accessible selectors, with the exception of the "results" div, which is only used to test the async dialog responses and isn't part of the actual component anyway

**Review by judithrocketmakers — APPROVED** [review]
Happy with how these have turned out - and given dialogs are tricky, very good job for a first go!

**Pull request merged**